File: server/src/custom-server/app/development/controllers/DevelopmentController.ts
import { CustomResult } from "../../../core/utils/ExpressUtil";
import ResultContent from "../../../core/utils/ResultContent";
import dayjs from "dayjs";

interface PingVo {
  dateTime: string;
  env: string;
}

export default {
  async ping(): Promise<CustomResult<PingVo>> {
    return new ResultContent<PingVo>()
      .withContent({
        dateTime: dayjs().toISOString(),
        env: "Development",
      })
      .withMessage("Teste")
      .build();
  },
};

File: backend/src/main/com/rot/development/controllers/DevelopmentController.kt
package com.rot.development.controllers

import com.rot.core.config.ApplicationConfig
import com.rot.core.jaxrs.Content
import com.rot.core.jaxrs.ResultContent
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import org.jboss.resteasy.reactive.RestResponse
import java.time.LocalDateTime

@ApplicationScoped
@Path("/api/core")
class DevelopmentController(
    val config: ApplicationConfig
) {
    @GET
    @Path("/ping")
    fun ping(): RestResponse<Content<MutableMap<String, Any?>>> {
        val metadata = mutableMapOf<String, Any?>()
        metadata["name"] = config.name()
        metadata["environment"] = config.environment()
        metadata["dateTime"] = LocalDateTime.now()

        return ResultContent.of(metadata)
            .withMessage("Ping server test successful")
            .build()
    }
}